src/Swift.Runtime/src/Swift/Data.cs

```diff
@@ -1,10 +1,119 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using Swift;
+using Swift.Runtime;
+
+
 /// <summary>
-/// Represents Foundation.Data type.
+/// Represents Swift Foundation.DataProtocol in C#.
+/// </summary>
+public interface ISwiftDataProtocol { }
+
+/// <summary>
+/// Represents Swift Foundation.ContiguousBytes in C#.
 /// </summary>
+public interface ISwiftContiguousBytes { }
 
-// TODO: https://github.com/dotnet/runtimelab/issues/2992
-public struct Data
+/// <summary>
+/// Represents Foundation.Data type.
+/// </summary>
+public struct Data : ISwiftObject
```

Comment:
Shouldn't the size be known at compile time?

Reply:
The struct layout should be known at compile time. If it contains a small payload, the data is inlined; otherwise, it represents a byte buffer in memory. Is that what you meant?

Reply:
I meant that the explicit size attribute is missing. Is it not necessary?

Reply:
Yes, it is lowered correctly without explicit layout.

Related to https://github.com/dotnet/runtimelab/pull/2964#discussion_r1932590598

> Yes, I believe that the Size is only important for nested structs so that we avoid any unwanted padding inside the struct. That said, I think we can keep it around until we figure out something better (e.g., https://github.com/dotnet/runtime/issues/100896).

/cc: @matouskozak